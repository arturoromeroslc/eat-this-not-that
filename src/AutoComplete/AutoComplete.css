.autocomplete__input {
	width: 100%;
	font-size: 18px;
	border: none;
	line-height: 22px;
	padding: 5px 10px 5px 25px;
	height: 32px;
}

.autocomplete__input:focus {
	outline: none;
}

.autocomplete__icon {
	display: inline-block;
	transform: rotate(-45deg);
	cursor: pointer;
	left: 1.6rem;
	position: absolute;
	color: #2060aa;
	font-size: 1.3rem;
}

.autocomplete__list {
	display: inline-flex;
	flex-direction: column;
	margin: 0;
	width: 100%;
	padding: 0;
	list-style-type: none;
	background: white;
}

.autocomplete__list-item {
	color: black;
}

.autocomplete__list-item:hover {
	background-color: lightgray;
	color: black;
	cursor: pointer;
}