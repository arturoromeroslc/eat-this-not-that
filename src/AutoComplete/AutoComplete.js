import axios from 'axios'
import Downshift from 'downshift'
import React, { Component } from 'react'
import PropTypes from 'prop-types';
import shortid from 'shortid'
import debounce from 'lodash.debounce'
import './AutoComplete.css'

const BASE_END_POINT = 'https://api.nutritionix.com/v2/autocomplete?q=';
const APP_ID = '&appId=be48f72d&appKey=36843f47de3c76347879e12f49cbfcf4';

export default class AutoComplete extends Component {
  constructor(props) {
    super(props)
    this.handleInputSearchChange = this.handleInputSearchChange.bind(this)
    this.handleSelectedItem = this.handleSelectedItem.bind(this)
    this.state = {
      items: []
    }
  }

  /**
   * get the value for the input
   * @param  {Object} e event object passed in
   */
  handleInputSearchChange(e) {
    const { value } = e.target

    if (value.length > 0) {
      this.getAutoCompleteResults(value)
      this.props.onChangedInputValue(value)
    }
    this.props.onChangedInputValue(value)
  }

  /**
   * when we click a list item call the prop onChange which will send an api call to get recipes
   * @param  {String} text item which was clicked
   */
  handleSelectedItem(text) {
    this.props.onChangedInputValue(text)
    this.props.onSelectedItem(text)
  }

  render() {
    return (
      <span>
        <span className="autocomplete__icon">âš²</span>
        <Downshift onChange={this.props.onSelectedItem}>
          {({
            selectedItem,
            getInputProps,
            getItemProps,
            highlightedIndex,
            isOpen,
          }) => (
            <div>
              <input
                className="autocomplete__input"
                placeholder="search"
                type="search"
                {...getInputProps({
                    onChange: (event) => {
                      const { value } = event.target
                      if (!value) {
                        return
                      }
                      debounce(
                        axios
                          .get(BASE_END_POINT + value + APP_ID)
                          .then((response) => {
                            const items = response.data.map(item => item.text)
                            this.setState({ items })
                          })
                          .catch((error) => {
                            console.error(error)
                          })
                      , 500
                    );
                    }
                  })}
              />
              {isOpen && (
              <div>
                {this.state.items.map((item, index) => (
                  <div
                    className="autocomplete__list"
                    key={shortid.generate()}
                    {...getItemProps({
                          item,
                          style: {
                            backgroundColor:
                              highlightedIndex === index ? 'black' : 'orange',
                            fontWeight: selectedItem === item ? 'bold' : 'normal',
                          },
                        })}
                  >
                    {item}
                  </div>
                    ))}
              </div>
                )}
            </div>
            )}
        </Downshift>
      </span>
    )
  }
}

AutoComplete.propTypes = {
  onChangedInputValue: PropTypes.func,
  onSelectedItem: PropTypes.func,
  value: PropTypes.string
}
