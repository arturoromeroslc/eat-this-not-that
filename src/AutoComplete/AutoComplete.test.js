import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AutoComplete from './AutoComplete'

global.requestAnimationFrame = function(callback) {
  setTimeout(callback);
};

test('AutoComplete snapshot test', () => {
  const component = shallow(<AutoComplete />)
})
