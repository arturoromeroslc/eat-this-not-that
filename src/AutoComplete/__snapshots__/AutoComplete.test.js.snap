exports[`test AutoComplete snapshot test 1`] = `
<span>
  <span
    className="autocomplete__icon">
    âš²
  </span>
  <input
    autoFocus={true}
    className="autocomplete__input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="search"
    type="search" />
  <ul
    className="autocomplete__list"
    style={Object {}} />
</span>
`;
